server/devpi_server/model.py

```diff
@@ -359,6 +371,32 @@ def verify_name(cls, indexname):
                 "indexname '%s' contains characters that aren't allowed. "
                 "Any ascii symbol besides -.@_ is blocked." % indexname)
 
+    def get_pypi_submit_principals(self, restrict_modify=None):
+        return self.stage.ixconfig.get("acl_upload", [])
+
+    def get_toxresult_upload_principals(self, restrict_modify=None):
+        return self.stage.ixconfig.get("acl_toxresult_upload", [':ANONYMOUS:'])
+
+    def get_index_delete_principals(self, restrict_modify=None):
+        if restrict_modify is None:
+            modify_principals = set(['root', self.stage.username])
+        else:
+            modify_principals = restrict_modify
+        return modify_principals
+
+    get_index_modify_principals = get_index_delete_principals
```

Comment:
Why is ``modify`` same as ``delete``. There should be a comment.

Reply:
Ok it is not that delete is same as modify. It is that they share permissions.

Reply:
Naming of those methods is maybe not obvious. I'd rather go for ``get_principals_for_index_delete``.

Reply:
Changed